cmd/clusterctl/client/cluster/topology.go

```diff
@@ -498,10 +498,18 @@ func getTemplates(objs []*unstructured.Unstructured) []*unstructured.Unstructure
 func (t *topologyClient) generateCRDs(objs []*unstructured.Unstructured) []*apiextensionsv1.CustomResourceDefinition {
```

Comment:
given that this func is getting more responsibilities, should we add a unit test ensuring that we handle properly CRD generations (and all its variants)?

Reply:
This function now is back to doing the simple task that it was doing before (ref: https://github.com/kubernetes-sigs/cluster-api/pull/6270#pullrequestreview-905977479). This function does not deal with any different kinds of cases anymore. The overall test we have should be enough to cover this function.